src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -126,6 +126,9 @@ private struct ComInterfaceInstance
         /// </summary>
         private static ComWrappers? s_globalInstanceForMarshalling;
 
+        private static int s_instanceCounter;
+        private readonly int id = Interlocked.Increment(ref s_instanceCounter);
```

Comment:
This ID can overflow if somebody creates short-lived ComWrapper instances. Artificial test that hits the recycling problem should take just minutes. Real-world cases that hit the problem would likely take days.

I think this needs to be 64-bit; or use a different dispenser scheme (e.g. address of something unmanaged).